website/docs/developer/research/research-logging-libraries.md

```diff
@@ -0,0 +1,150 @@
+    }
+```
+
+Resulting Cloudwatch message:
+```
+{
+    "level": "ERROR",
+    "location": "some_func:23",
+    "message": "Manual logging: ''some key error''",
+    "timestamp": "2022-09-01 20:11:25,278+0000",
+    "service": "powertools_test_service_name",
+    "cold_start": true,
+    "function_name": "lambda_name",
+    "function_memory_size": "128",
+    "function_arn": "arn:aws:lambda:us-west-2:236859827343:function:lambda_name",
+    "function_request_id": "bdb3bb29-5023-498a-b7a5-0582b03e215e",
+    "xray_trace_id": "1-631111ec-64c294e313312cb53eb68777"
+}
+```
+
+## Recommendation
+- As the [Python CumulusLogger](https://github.com/nasa/cumulus-message-adapter-python#logging-with-cumuluslogger) is tied to Cumulus, 
+  it is recommended that we decouple from it.
+  - Any references to deployed layer should be removed from Terraform, including deployed layer and Step Function configuration.
+  - Schema validation checks should be added to calling function.
+  - References to run_cumulus_task should be replaced with manual transformation of inputs into non-architecture-specific formats.
+  - `copy_to_glacier` S3 requirements could be accomplished with a helper Lambda developed initially by us and maintained by Cumulus that handles the S3 retrieval 
+    and passes the result along to `copy_to_glacier`.
+- Given the standardized implementation and features of Powertools, it is a good pick for replacing the CumulusLogger in Orca code.
+- Recommend installing via Pip install as opposed to AWS Layer.
+  [Documentation indicates](https://awslabs.github.io/aws-lambda-powertools-python/latest/core/logger/#removing-additional-keys) that there can be "unintended side effects if you use Layers" with certain features.
+- Since Powertools properly implements the [standard Logging library](https://docs.python.org/3/library/logging.html), we can now use the `logging.exception` method to auto-capture and log exceptions, including stack traces.
+  From the "Auto logging" code example above, Cloudwatch message is:
+  ```
+  {
+      "level": "ERROR",
+      "location": "some_func:22",
+      "message": "Auto logging",
+      "timestamp": "2022-09-01 20:11:25,278+0000",
+      "service": "powertools_test_service_name",
+      "cold_start": true,
+      "function_name": "lambda_name",
+      "function_memory_size": "128",
+      "function_arn": "arn:aws:lambda:us-west-2:236859827343:function:lambda_name",
+      "function_request_id": "bdb3bb29-5023-498a-b7a5-0582b03e215e",
+      "exception": "Traceback (most recent call last):\n  File \"/var/task/powertools_test.py\", line 20, in some_func\n    raise KeyError(\"some key error\")\nKeyError: 'some key error'",
+      "exception_name": "KeyError",
+      "xray_trace_id": "1-631111ec-64c294e313312cb53eb68777"
+  }
+  ```
+- Powertools also contains a [Tracing library](https://awslabs.github.io/aws-lambda-powertools-python/latest/core/tracer/) which could be helpful in debugging. Presently untested.
```

Comment:
Since the recommendation is to use AWS power tools, it may be good here at the bottom to add links to the cards for either/planning or implementation of the replacing CMA with poser tools in our code. Thoughts?

Reply:
Sounds good. If the plan sounds reasonable so far, I'll get them written up and added.

Reply:
Added cards